src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size = 0;
+    private T[] arr;
+    private T[] newArr;
+
+    public ArrayList() {
+        arr = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size < arr.length) {
+            arr[size] = value;
+            size++;
+        } else {
+            resize();
```

Comment:
Don't complicate if-else construction. It's better to call 'resize()' method without the 'else' block since you're doing the same operations after resizing in both cases.